src/api/sendGrid.ts

```diff
@@ -0,0 +1,23 @@
+const sgMail = require('@sendgrid/mail');
+const template_id = 'd-2e3aabdb9a3c45a393877596534c57c2'; // temporary holder, change to required template id
+
+export interface Personalizations {
+    to: string|string[];
+    dynamic_template_data?: Record<string, string>;
+}
+
+export const sendEmail = async (personalizations: Personalizations[]) => {
+
+    sgMail.setApiKey(process.env.SENDGRID_API_KEY);
+
+    const message = {
+        personalizations: personalizations,
+        from: 'yizhang1728@gmail.com', // yuanjian@163
+        template_id: template_id,
+    }
+
+    sgMail
```

Comment:
avoid using `.then`. use await:
```
console.log(`Sending email: ${personalization}`)
await mail.send()
```

Let upper stacks catch the error. Do not "swallow" the error here at the lower stack -- this is an anti-pattern against "fail fast and fail safe".